param.go

```diff
@@ -277,9 +223,29 @@ func (po paramObject) DotParam() []*dot.Param {
 	return types
 }
 
+func getParamOrder(gh *graphHolder, param param) []int {
+	var orders []int
+	switch p := param.(type) {
+	case paramSingle:
+		providers := gh.c.getValueProviders(p.Name, p.Type)
+		for _, provider := range providers {
+			v := gh.orders[key{t: provider.CType()}]
+			orders = append(orders, v)
+		}
+	case paramGroupedSlice:
+		v := gh.orders[key{t: p.Type, group: p.Group}]
+		orders = append(orders, v)
```

Comment:
Is this right? It seems significantly different from the caller's handling of paramGroupedSlice.

https://github.com/uber-go/dig/pull/301/files#diff-a27f75d4c63afb2160bde845e6453abce79e88c5ee81d146711382455f05c1e3R352-R355

I also can't tell for sure if this will actually happen, to be honest.
I _think_ that this is resolving the dependency from function argument to value group node,
but it's a bit unclear.
If that's right, _maybe_ separating paramGroupedSlice from valueGroupNode might make that clearer?

Reply:
@sywhang I might be misunderstanding some of the code here. Is this not a concern?

Reply:
talked offline on Slack but for records:

these two are doing different things: EdgesFrom tries to get the orders of the dependencies of a paramGroupedSlice. 
getParamOrder tries to get the order of a paramGroupedSlice from the graph.